apps/sms/views/shared/js/utils.js

```diff
@@ -711,6 +711,62 @@
     },
 
     /**
+     * Returns a rate limited function, useful to reduce the number of function
+     * calls when listening to user input (keypress) or browser events.
```

Comment:
nit: About the edge case that preventFisrCall/preventLastCall both be true. I think it's enough to add some comments here. But it's very optional :)